### PR TITLE
refactor: add lightweight styling resources for ProgressRing control

### PR DESCRIPTION
**Pull request by Arieldelossantos**
﻿GitHub Issue: #https://github.com/unoplatform/Uno.Themes/issues/993

## PR Type

What kind of change does this PR introduce?

- Refactoring (no functional changes, no api changes)

## Description

Added Lightweight styling resources for ProgressRing control

||After modifying **ColorPaletteOverride.xaml** file|
|---|---|
|Screenshot_1685474980|Screenshot_1685501455|

### CODE REVIEW

File: src/library/Uno.Material/Styles/Controls/v2/ProgressRing.xaml [hunk 1]
```diff
@@ -10,28 +10,42 @@
...
-							   Value="{StaticResource M3MaterialDeterminateAnimation_Uno}" />
+						Value="{StaticResource M3MaterialDeterminateAnimation_Uno}" />
 		<not_win:Setter Property="IndeterminateSource"
-							   Value="{StaticResource M3MaterialIndeterminateAnimation_Uno}" />
+						Value="{StaticResource M3MaterialIndeterminateAnimation_Uno}" />
 		<Setter Property="Foreground"
-				Value="{ThemeResource PrimaryBrush}" />
+				Value="{ThemeResource ProgressRingForeground}" />
 		<Setter Property="Background"
 				Value="Transparent" />
```
Comment (kazo0) [inline src/library/Uno.Material/Styles/Controls/v2/ProgressRing.xaml:50]: Might as well extract a `ProgressRingBackground` while we're at it

File: src/library/Uno.Material/Styles/Controls/v2/ProgressRingWinUI.xaml [hunk 2]
```diff
@@ -18,22 +18,34 @@
...
 		        Value="Transparent" />
 	</not_win:Style>
 
 	<win:Style x:Key="MaterialProgressRingStyle"
 	       TargetType="muxc:ProgressRing">
 		<Setter Property="Foreground"
-		        Value="{ThemeResource PrimaryBrush}" />
+				Value="{ThemeResource ProgressRingForeground}" />
 		<Setter Property="Background"
 		        Value="Transparent" />
```
Comment (kazo0) [inline src/library/Uno.Material/Styles/Controls/v2/ProgressRingWinUI.xaml:50]: Same thing here, `ProgressRingBackground`

File: src/library/Uno.Material/Styles/Controls/v2/ProgressRing.xaml (hunk 1 above)
Comment (Arieldelossantos) [reply src/library/Uno.Material/Styles/Controls/v2/ProgressRing.xaml:50]: but isn't the bg always be transparent??

Comment (kazo0) [reply src/library/Uno.Material/Styles/Controls/v2/ProgressRing.xaml:50]: Yes, so it'll be preset to be an alias to the `SystemControlTransparentBrush` but someone may want to customize the brush to be something else, especially later down the line when we may add visual state customization to the ProgressRing

Comment (Arieldelossantos) [reply src/library/Uno.Material/Styles/Controls/v2/ProgressRing.xaml]: done

File: src/library/Uno.Material/Styles/Controls/v2/ProgressRingWinUI.xaml [hunk 3]
```diff
@@ -18,24 +18,39 @@
...
+						ResourceKey="SystemControlTransparentBrush" />
+	</ResourceDictionary>
+
+	<ResourceDictionary x:Key="Light">
+		<!--MaterialProgressRingStyle-->
+		<StaticResource x:Key="ProgressRingForeground"
+						ResourceKey="PrimaryBrush" />
+		<StaticResource x:Key="ProgressRingBackground"
+						ResourceKey="SystemControlTransparentBrush" />
+	</ResourceDictionary>
```
[the remark this answers is not in the document]
Comment (Arieldelossantos) [reply src/library/Uno.Material/Styles/Controls/v2/ProgressRingWinUI.xaml]: fixed

**Pull request merged**